### PR TITLE
[fix](jdbc catalog) fix jdbc-connector coredump as get env return nullptr

### PR DESCRIPTION
**Pull request by zy-kkk**
## Proposed changes

Issue Number: close #xxx

```
Call to AttachCurrentThread failed with error: -1
getJNIEnv: getGlobalJNIEnv failed
*** Query id: fcd93e45c5df4271-95179a8e8523a9df ***
*** is nereids: 1 ***
*** tablet id: 0 ***
*** Aborted at 1710361477 (unix time) try "date -d @1710361477" if you are using GNU date ***
*** Current BE git commitID: 2f4401189a ***
*** SIGSEGV address not mapped to object (@0x0) received by PID 90049 (TID 97639 OR 0x7f83b1cb9700) from PID 0; stack trace: ***
 0# doris::signal::(anonymous namespace)::FailureSignalHandler(int, siginfo_t*, void*) at /home/zcp/repo_center/doris_branch-2.1/doris/be/src/common/signal_handler.h:421
 1# 0x00007F8BCF8770A7 in /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/amd64/server/libjvm.so
 2# JVM_handle_linux_signal in /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/amd64/server/libjvm.so
 3# 0x00007F8BCF87002C in /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/amd64/server/libjvm.so
 4# 0x00007F8BD7C38090 in /lib/x86_64-linux-gnu/libc.so.6
 5# JNIEnv_::DeleteGlobalRef(_jobject*) at /usr/lib/jvm/java-8-openjdk-amd64/include/jni.h:848
 6# doris::vectorized::JdbcConnector::close(doris::Status) at /home/zcp/repo_center/doris_branch-2.1/doris/be/src/vec/exec/vjdbc_connector.cpp:88
 7# doris::vectorized::NewJdbcScanner::close(doris::RuntimeState*) at /home/zcp/repo_center/doris_branch-2.1/doris/be/src/vec/exec/scan/new_jdbc_scanner.cpp:207
```

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zy-kkk** [comment]
run buildall

**Review by morningman — DISMISSED** [review]
LGTM

File: be/src/vec/exec/vjdbc_connector.cpp [hunk 1]
```diff
@@ -83,8 +83,12 @@ Status JdbcConnector::close(Status /*unused*/) {
     if (_is_in_transaction) {
         RETURN_IF_ERROR(abort_trans());
     }
-    JNIEnv* env;
-    RETURN_IF_ERROR(JniUtil::GetJNIEnv(&env));
+    JNIEnv* env = nullptr;
+    Status status = JniUtil::GetJNIEnv(&env);
```
Comment (yiguolei) [inline be/src/vec/exec/vjdbc_connector.cpp:87]: There are many APIs that call GetJNIEnv, and use RETURN_IF_ERROR(JniUtil::GetJNIEnv(&env));
I think we should add check in GetJNIEnv if the result == nullptr, should return an error status.

**Review by morningman — DISMISSED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]

**Comment by yiguolei** [comment]
run buildall

**Review by morningman — APPROVED** [review]

**Pull request merged**